src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp

```diff
@@ -52,7 +53,21 @@ class SinglePhasePoromechanics : public PoromechanicsSolver< FLOW_SOLVER, MECHAN
    * @brief name of the node manager in the object catalog
    * @return string that contains the catalog name to generate a new SinglePhasePoromechanics object through the object catalog.
    */
-  static string catalogName();
+  template< typename _FLOW_SOLVER=FLOW_SOLVER >
+  static
+  typename std::enable_if< std::is_same< _FLOW_SOLVER, SinglePhaseBase >::value, string >::type
+  catalogName()
+  {
+    return "SinglePhasePoromechanics";
+  }
+  template< typename _FLOW_SOLVER=FLOW_SOLVER >
+  static
+  typename std::enable_if< std::is_same< _FLOW_SOLVER, SinglePhaseReservoirAndWells< SinglePhaseBase > >::value, string >::type
+  catalogName()
+  {
+    return SinglePhaseReservoirAndWells< SinglePhaseBase >::catalogName() + "Poromechanics";
+  }
```

Comment:
```suggestion
 static string catalogName()
  {
      if constexpr ( std::is_same_v< FLOW_SOLVER, SinglePhaseBase > )
      {
          return "SinglePhasePoromechanics";
      }
      else if constexpr( std::is_same_v< FLOW_SOLVER, SinglePhaseReservoirAndWells< SinglePhaseBase > > )
      {
          return SinglePhaseReservoirAndWells< SinglePhaseBase >::catalogName() + "Poromechanics";
      }
      else
      {
          static_assert(false, "blah blah");
      }
  }
```
Is this working?

Reply:
i took it from here to be honest: https://github.com/GEOS-DEV/GEOS/blob/ddf3e97a398c1261cae81cf2f864112061e5b259/src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseFVM.hpp#L101
can you your version if it's preferred

Reply:
If my version is working and doing the right thing, I think it's more modern and simple.
I think that the `if constexpr` c++17 feature was thought for this use case (at least).

Reply:
that `static_assert` fails for some reason, not sure why

Reply:
My `static_assert` code was bad, I wrote it very fast and surely you need to be something a little smarter!

Reply:
i put together something, let's see if it can pass the tests